src/coffea/nanoevents/methods/nanoaod.py

```diff
@@ -13,7 +13,14 @@
 
 class _NanoAODEvents(behavior["NanoEvents"]):
     def __repr__(self):
-        return f"<event {self.run}:{self.luminosityBlock}:{self.event}>"
+        if (
+            hasattr(self, "run")
+            and hasattr(self, "luminosityBlock")
+            and hasattr(self, "event")
+        ):
```

Comment:
Another option may be to use getattr(self, <field>, "??") such that if 1 or 2 of the fields are there, they'll still print. (Or something similar)

Reply:
repeated warnings tend to be annoying to users, for what it's worth.